core/WFCORE-1427_Convenient_declaration_of_server_suspend_timeout.adoc

```diff
@@ -0,0 +1,129 @@
+* The host level operation 'suspend-servers' will have a timeout parameter named 'suspend-timeout' that controls how long the suspend operation should wait for in-flight requests to complete before returning.
+* The timeout is per server, not an aggregate overall timeout for operations that affect multiple servers.
+* Host controllers will send the suspend operation to the managed servers in parallel.
+* There are already operations which use a timeout parameter for similar purposes. The name of this parameter leads to confusion because it can be interpreted as an operation timeout. In order to have a consistency and to fix this confusion, this 'timeout' parameter will be deprecated in all relevant operations. A 'suspend-timeout' parameter will be added to be used instead of the deprecated one.
+
+=== Nice-to-Have Requirements
+* HAL support. This should be a hard requirement for a subsequent release but is a nice to have initially.
+* The use of 'timeout' param of the high level shutdown operation and adding a 'timeout' param to the low level 'shutdown' host operation in domain mode to get the host level graceful shutdown not oly via the suspend operation, but also using the shutdown operation.
+
+
+=== Non-Requirements
+* For host level 'suspend-servers' operation affecting multiple servers, treating the 'timeout' as being some overall timeout. The timeout applies to each server.
+
+=== Design notes
+
+These are just examples of how the new operations looks like:
+
+- If the user wants to reload a host restarting its servers gracefully:
+
+[source]
+--
+[domain@localhost:9990 /] /host=master:suspend-servers(timeout=20)
+[domain@localhost:9990 /] /host=master:reload(restart-servers=true)
+--
+
+
+- If the user wants to shutdown gracefully a host
+
+[source]
+--
+[domain@localhost:9990 /] /host=master:suspend-servers(timeout=20)
+[domain@localhost:9990 /] /host=master:shutdown()
+--
+
+The 'timeout' parameter will be deprecated in the following operations and 'suspend-timeout' parameter will be added to replace its function:
+
+[source]
+--
+[domain@localhost:9990 /] :stop-servers(timeout=X)
+[domain@localhost:9990 /] /host=*/server=*:stop(timeout=X)
+[domain@localhost:9990 /] /server-group=*:stop-servers(timeout=X)
+[domain@localhost:9990 /] :suspend-servers(timeout=X)
+[domain@localhost:9990 /] /host=*/server=*:suspend(timeout=X)
+[domain@localhost:9990 /] /server-group=*:suspend-servers(timeout=X)
+
+[standalone@localhost:9990  /] :suspend(timeout=X)
+[standalone@localhost:9990  /] :shutdown(timeout=X)
+[standalone@localhost:9990  /] shutdown --timeout
+--
+
+Notice the following: +
+{issue-base-url}/WFCORE-1917[WFCORE-1917] _stop_ and _suspend_ operations were deprecated under +/host=*/server-config=*+ resource. We are not going to mofify the 'timeout' attribute on those operations. +
+{issue-base-url}/WFCORE-3804[WFCORE-3804] _stop_ and _suspend_ operations will be available under +/host=*/server=*+ resource once this issue is merged.
```

Comment:
Well, which operations then? Here is stated that `stop` and `suspend` but in the jira title and description, there is mentioned `reload` and `suspend` operations instead. I think that those operations mentioned in jira are correct.

Reply:
Hm, now I realized that effectively this should be true for all three operations, right?